### ISSUE TITLE
Link errors in pod files

### ISSUE THREAD

**Issue by finanalyst** [Raku/doc#3833]
## Links that have errors

After using `Collection` on the Raku documentation files, I created a test to check all links, that is POD6 with the FormatCode `L<some title|url>`

There are a variety of errors. I am still developing the test. 

However, the most urgent test are the 404 and no-longer existing domains. Please refer to [Error Report](https://raku.finanalyst.org/error-report.html).

The non-existent files mainly refer to files generated by Documentable, but which do not exist under `doc/` in the repo. 

There are other errors, such as non-existent targets in existing files. 

## Suggestions

The 404 and non-existent domains can be removed from the Raku documentation. I doubt whether any of these references are needed.

**Comment by JJ** [comment]
Related: #561

**Comment by Altai-man** [comment]
Are URLs under `Links to missing files` section considered broken? If so, the heuristic does not seem right, e.g.

>/routine/pick
>/type/Submethod

and others similar - they exist and work, it seems there are numerous false positives in this list.

However, the funny case of `/language/type/Str` was detected (I detected it with my tooling as well), so there is certainly some truth in.

**Comment by finanalyst** [comment]
@Altai-man 
I draw attention to the first section. I am working on the tests for the other sections. The tests have quite a few "false negatives", as you point out.
Most (but not all) the missing file links are due to Secondary files, that is files which do not have a direct `doc/..../fn.pod6` file. They are generated by Documentable. The approach was first introduced when Moritz was developing the site in order to increase speed of searching (I think).

But the practice is very bad in my opinion. Changes in Primary files will have an effect on secondary files, so links to secondary files will be changed.

**Comment by finanalyst** [comment]
I have now created PRs to remove all 404 and unknown host errors.  All these changes have minimal impact on the rest of the documentation. 

When those PRs are merged, I think this issue can be closed.

When I have a better error detection test for 'Non-existent links in existing files', I will open another issue. There are many true negatives due to bit-rot or typos, but too many false negatives.

The issue of Secondary files is really one of Policy. I have created a PR several years ago to remove links to secondary files in primary files. But the changes are extensive in number.
